code/headers/ssd1306.hpp

```diff
@@ -94,53 +94,103 @@ namespace r2d2::display {
+        void command(ssd1306_command c) {
+            // create command packet
+            uint8_t data[] = {ssd1306_cmd_prefix, (uint8_t)c};
+
+            // write command to the bus
+            bus.write(address, data, sizeof(data) / sizeof(uint8_t));
+        }
 
         /// send a command with one data byte
-        void command(ssd1306_command c, uint8_t d0);
+        void command(ssd1306_command c, uint8_t d0) {
+            // create command packet
+            uint8_t data[] = {ssd1306_cmd_prefix, (uint8_t)c, ssd1306_cmd_prefix,
+                            d0};
+
+            // write command to the bus
+            bus.write(address, data, sizeof(data) / sizeof(uint8_t));
+        }
 
         /// send a command with two data bytes
-        void command(ssd1306_command c, uint8_t d0, uint8_t d1);
+        void command(ssd1306_command c, uint8_t d0, uint8_t d1) {
+            // create command packet
+            uint8_t data[] = {ssd1306_cmd_prefix, (uint8_t)c,
+                            ssd1306_cmd_prefix, d0,
+                            ssd1306_cmd_prefix, d1};
+
+            // write command to the bus
+            bus.write(address, data, sizeof(data) / sizeof(uint8_t));
+        }
 
         /// send one byte of data
-        void data(uint8_t d);
+        void data(uint8_t d) {
+            // create data packet
+            uint8_t data[] = {ssd1306_data_prefix, d};
+
+            // write data to the bus
+            bus.write(address, data, sizeof(data) / sizeof(uint8_t));
+        }
 
         /// write the pixel byte d at column x page y
-        void pixels_byte_write(hwlib::xy location, uint8_t d);
+        void pixels_byte_write(hwlib::xy location, uint8_t d) {
+            // check if we need to update the current cursor of the screen
+            if (location != cursor) {
+                command(ssd1306_command::column_addr, location.x, 127);
+                command(ssd1306_command::page_addr, location.y, 7);
+                cursor = location;
+            }
+
+            // write data to the screen
+            data(d);
+
+            // update the local cursor
+            cursor.x++; // TODO: cursor fallthrough
+        }
 
     public:
         /**
          * @brief width of display
          *
          */
-        constexpr static uint8_t width = 128;
+        constexpr static uint8_t width = Display_Size_Width;
 
         /**
          * @brief height of display
          *
          */
-        constexpr static uint8_t height = 64;
+        constexpr static uint8_t height = Display_Size_Height;
 
         /**
          * @brief converts a hwlib::color to the pixel data for the screen with
          * a maximum of two bytes for every pixel
          *
          * @param c
          */
-        uint16_t color_to_pixel(const hwlib::color &c) override;
+        uint16_t color_to_pixel(const hwlib::color &c) override {
+            // return as bool becouse we only need bools for the display
```

Comment:
spelling mistake "Because"

Reply:
This should have been fixed after this pr, https://github.com/R2D2-2019/display/pull/22#discussion_r278220847 however this has not happened.

Reply:
I see now that this has happend here https://github.com/R2D2-2019/display/issues/23.

Reply:
Also, it should be "because" without the capital "B".

Reply:
Fixed in 536b13e